From 360a40f8a83e08a09b34cb12e269c793028b315e Mon Sep 17 00:00:00 2001
From: Adheer Chandravanshi <adheer.chandravanshi@qlogic.com>
Date: Tue, 17 Sep 2013 08:07:28 -0400
Subject: [PATCH] flashnode: Add support to set ISCSI_FLASHNODE_CHAP_OUT_IDX
 param

Add support to set the chap_out_idx session param of flashnode
entry.

Signed-off-by: Adheer Chandravanshi <adheer.chandravanshi@qlogic.com>
Signed-off-by: Vikas Chaudhary <vikas.chaudhary@qlogic.com>
---
 usr/flashnode.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/usr/flashnode.c b/usr/flashnode.c
index da1392a..fe5ab57 100644
--- a/usr/flashnode.c
+++ b/usr/flashnode.c
@@ -449,6 +449,12 @@ int flashnode_build_config(struct list_head *params,
 			    fnode->sess.portal_type,
 			    sizeof(fnode->sess.portal_type)))
 				count++;
+		} else if (!strcmp(param->name,
+			  to_key(FLASHNODE_SESS_CHAP_OUT_IDX))) {
+			if (!flashnode_fill_uint32(fnode, &iov[count],
+			    ISCSI_FLASHNODE_CHAP_OUT_IDX,
+			    fnode->sess.chap_out_idx))
+				count++;
 		} else if (!strcmp(param->name, to_key(FLASHNODE_CONN_PORT))) {
 			if (fnode->conn[0].port)
 				port = fnode->conn[0].port;
-- 
1.8.3.1

